### PR TITLE
Traefik reverse proxy

### PR DESCRIPTION
**Pull request by zerpex**
Hi, 

Here is a sample conf for using your mailserver with traefik.

Regards,
z.

### CODE REVIEW

**Comment by hardware** [comment]
Thanks. I need some more time to test web security guidelines listed on :

* https://www.hardenize.com/
* https://www.ssllabs.com/ssltest
* https://securityheaders.io
* https://observatory.mozilla.org

with Traefik reverse proxy. 

Wonderfall's boring-nginx had some security features that traefik did not have enabled by default.  I will look [unrolled/secure](https://github.com/unrolled/secure), Traefik use this middleware to manage the web security layer.

**Comment by zerpex** [comment]
What exactly do you need ?
Maybe I can help.

**Comment by ksylvan** [comment]
I can look this over during my upcoming break as well.

**Comment by hardware** [comment]
> What exactly do you need ?

I will use Traefik not only for this docker image but also for a forum that I maintain in parallel. This forum need some specific frontend rules to determine how incoming requests are forwarded to the backend. I have to understand how everything works before to push this.

And I'll be on vacation for Christmas holidays.

> Maybe I can help.

You can try to get a good score on the websites I mentioned above and share your configuration in this pull request 😃 

Happy Christmas 🎅🎄

File: traefik.toml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  # caServer = "https://acme-staging.api.letsencrypt.org/directory"
+  email = "me@domain.tld"
+  storage = "/etc/traefik/acme/acme.json" 				# or "traefik/acme/account" if using KV store
+  entryPoint = "https"
+  onDemand = false
+  OnHostRule = true
+
+[docker]
+  endpoint = "unix:///var/run/docker.sock"
+  domain = "toolz.pm"
```
Comment (ksylvan) [inline traefik.toml]: Shouldn't this be `domain.tld` and not `toolz.pm`?

Comment (hardware) [reply traefik.toml]: https://github.com/hardware/mailserver/blob/master/traefik.sample.toml#L22 ;)

Comment (ksylvan) [reply traefik.toml]: Cool. Thanks.

**Pull request merged**